client/post-editor/editor-categories-tags/accordion.jsx

```diff
@@ -93,37 +73,33 @@ module.exports = React.createClass( {
 				/>
 			</AccordionSection>
 		);
-	},
+	}
 
 	getCategoriesSubtitle() {
-		const siteId = this.props.site.ID;
+		const { translate, postTerms, defaultCategory } = this.props;
+		const categories = get( postTerms, 'category' );
 
-		// There's also post.categories but it doesn't have category names
-		// either, so we need to get them from the site categories data
-		const categories = this.getCategories().map( ( categoryId ) => {
-			const category = TermStore.get( siteId, categoryId );
-			return category ? unescapeString( category.name ) : null;
-		} ).filter( categoryName => categoryName );
+		const categoryNames = map( categories, ( category ) => {
+			return unescapeString( category.name );
+		} );
+		const categoryNamesCount = size( categoryNames );
 
-		switch ( categories.length ) {
+		switch ( categoryNamesCount ) {
 			case 0:
-				return null; // No categories subtitle
+				return defaultCategory ? defaultCategory.name : null;
```

Comment:
How are we guaranteed that the default category will be loaded? Should we maybe render a `<QueryTerms />` component specifically targeting the default category?